src/mumble/GlobalShortcut_linux.cpp

```diff
@@ -0,0 +1,187 @@
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "GlobalShortcut_linux.h"
+
+#include "Settings.h"
+
+#include <QtCore/QFileSystemWatcher>
+#include <QtCore/QSocketNotifier>
+
+#ifdef Q_OS_LINUX
+#	include <fcntl.h>
+#	include <linux/input.h>
+#endif
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name
+// (like protobuf 3.7 does). As such, for now, we have to make this our last incl
+#include "Global.h"
+
+// We have to use a global 'diagnostic ignored' pragmas because
+// we still support old versions of GCC. (FreeBSD 9.3 ships with GCC 4.2)
+#if defined(__GNUC__)
+// ScreenCount(...) and so on are macros that access the private structure and
+// cast their return value using old-style-casts. Hence we suppress these warnings
+// for this section of code.
+#	pragma GCC diagnostic ignored "-Wold-style-cast"
+// XKeycodeToKeysym is deprecated.
+// For backwards compatibility reasons we want to keep using the
+// old function as long as possible. The replacement function
+// XkbKeycodeToKeysym requires the XKB extension which isn't
+// guaranteed to be present.
+#	pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+
+/**
+ * Returns a platform specific GlobalShortcutEngine object.
+ *
+ * @see GlobalShortcutX
+ * @see GlobalShortcutLinux
+ * @see GlobalShortcutMac
+ * @see GlobalShortcutWin
+ */
+GlobalShortcutEngine *GlobalShortcutEngine::platformInit() {
+	return new GlobalShortcutLinux();
+}
+
+GlobalShortcutLinux::GlobalShortcutLinux() {
+	if (g.s.bEnableEvdev) {
+		QString dir             = QLatin1String("/dev/input");
+		QFileSystemWatcher *fsw = new QFileSystemWatcher(QStringList(dir), this);
+		connect(fsw, SIGNAL(directoryChanged(const QString &)), this, SLOT(directoryChanged(const QString &)));
+		directoryChanged(dir);
+
+		if (qsKeyboards.isEmpty()) {
+			foreach (QFile *f, qmInputDevices)
+				delete f;
+			qmInputDevices.clear();
+
+			delete fsw;
+			qWarning("GlobalShortcutLinux: Unable to open any keyboard input devices under /dev/input");
+		} else {
+			return;
+		}
+	} else {
+		qWarning("GlobalShortcutLinux: No evdev support, global shortcuts disabled");
+	}
+}
+
+GlobalShortcutLinux::~GlobalShortcutLinux() {
+	wait();
+}
+
+// One of the raw /dev/input devices has ready input
+void GlobalShortcutLinux::inputReadyRead(int) {
+	if (!g.s.bEnableEvdev) {
+		return;
+	}
+
+	struct input_event ev;
+
+	if (bNeedRemap)
+		remap();
+
+	QFile *f = qobject_cast< QFile * >(sender()->parent());
+	if (!f)
+		return;
+
+	bool found = false;
+
+	while (f->read(reinterpret_cast< char * >(&ev), sizeof(ev)) == sizeof(ev)) {
+		found = true;
+		if (ev.type != EV_KEY)
+			continue;
+		bool down;
+		switch (ev.value) {
+			case 0:
+				down = false;
+				break;
+			case 1:
+				down = true;
+				break;
+			default:
+				continue;
+		}
+		int evtcode = ev.code + 8;
```

Comment:
Where is the `+8` coming from? The constant either needs a descriptive name and/or we need an explanatory comment